src/cryptography/hazmat/primitives/ciphers/modes.py

```diff
@@ -210,12 +210,17 @@ def __init__(
             utils._check_bytes("tag", tag)
             if min_tag_length < 4:
                 raise ValueError("min_tag_length must be >= 4")
-            if len(tag) < min_tag_length:
+            tag_len = len(tag)
+            if tag_len < min_tag_length:
                 raise ValueError(
                     "Authentication tag must be {} bytes or longer.".format(
                         min_tag_length
                     )
                 )
+            elif tag_len > 16:
```

Comment:
Why not use `self._block_size_bytes`?

Reply:
The GCM class doesn't have that value because it doesn't actually know the block size until it's composed with an algorithm. That said, it's only defined for 128-bit block ciphers.

Reply:
Oh duh, can we put this check in `validte_for_algorithm?`

Reply:
We could, although then we can have tag errors in two places. I'm fine with that though.